### PR TITLE
fix: prevent disabled Tag from being removed with keyboard nav

### PR DESCRIPTION
**Pull request by minzzang144**
Closes https://github.com/adobe/react-spectrum/issues/7080

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/@react-aria/tag/src/useTag.ts [hunk 1]
```diff
@@ -73,34 +99,53 @@ export function useTag<T>(props: AriaTagProps<T>, state: ListState<T>, ref: RefO
...
+      isDisabled,
       id: buttonId,
-      onPress: () => onRemove ? onRemove(new Set([item.key])) : null,
-      excludeFromTabOrder: true
+      onPress: () => (onRemove ? onRemove(new Set([item.key])) : null),
+      excludeFromTabOrder: true,
     },
     rowProps: mergeProps(rowProps, domProps, linkProps, {
-      tabIndex: (isFocused || state.selectionManager.focusedKey == null) ? 0 : -1,
+      tabIndex: getTabIndex(isDisabled, isFocused, isNonFocused),
```
Comment (minzzang144) [inline packages/@react-aria/tag/src/useTag.ts:140]: Thank you for the opportunity to resolve this issue.
I solved an issue where the TabGroup component was deleted when it was disalbed.
I don't think it's a good code, but if there is a better code or something wrong, please point it out. thanks!

**Review by snowystinger — COMMENTED** [review]
Thanks for the PR. It looks like you've run prettier or something equivalent over the file. Would you mind undoing that so that we can review the actual change? right now there is a bunch of noise and things that will fail our lint. Thanks again!

**Comment by minzzang144** [comment]
@snowystinger Thank you for your help. Edit completed!

File: packages/@react-aria/tag/src/useTag.ts [hunk 2]
```diff
@@ -80,19 +85,31 @@ export function useTag<T>(props: AriaTagProps<T>, state: ListState<T>, ref: RefO
...
+    isDisabled: boolean,
+    isFocused: boolean,
+    isNonFocused: boolean
+  ) {
+    if (isDisabled) {
+      return -1;
+    }
+    return isFocused || isNonFocused ? 0 : -1;
+  }
+
```
Comment (snowystinger) [inline packages/@react-aria/tag/src/useTag.ts:99]:
```suggestion
  let tabIndex = -1;
  f (!isDisabled && (isFocused || state.selectionManager.focusedKey == null)) {
    tabIndex = 0;
  }
```

We can just derive it as part of render, no need to add an additional function call with arguments we already have.

**Review by snowystinger — COMMENTED** [review]
Thanks for the quick update!

Would you mind adding a test for this?

**Review by snowystinger — APPROVED** [review]
Thanks for having a go at that. So that you know why I changed it, I focused on the behavior instead of the implementation. So interacted with the test in the same way a user would in the real world since it was reproducible there.
It was a good start though and I appreciate it. Looks good now.

**Comment by minzzang144** [comment]
Oh... you tested it based on the component where the hook is used, not the hook itself. I think this method looks much better too! I had to look at the code to understand it. Thank you for giving me such a great opportunity :)

**Review by reidbarber — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**